### PR TITLE
[OSPP]The proposal for TinyMS support in Python SDK

### LINKED ISSUE #307: [Feature] Implement Sedna Python SDK support for MindSpore and TinyMS
### Background:
1. Sedna is an edge-cloud collaboration tool that enables collaborative training and collaborative reasoning across edge-cloud. 
2. MindSpore is a domestic mainstream AI computing framework for all scenarios of device-edge-cloud, providing end-to-end capabilities for model development, model operation, and model deployment of all-scenario AI. 
3. TinyMS is a high-level API tool developed based on MindSpore.

### Why is this project needed:
1. Currently Sedna only supports sklearn xgboost, Tensorflow, and Keras. 
2. This project aims to implement Sedna Python SDK support for MindSpore and MindSpore high-level API tool TinyMS based on Sedna lib development interface.

### What contents are to be added/modified:
*You can refer to Sedna's lib development interface for the implementation of TensorFlow and Keras.*
- Deliverables:
 1. Implement the code, design the demo, run the demo, and submit the pr
 2. Related documentation: usage documentation, use case reports 
- Qualified Standard:
 1. Code standards comply with prescribed open source protocols and secure coding standards
 2. Successfully run the code and pass the test case
 3. Successfully merged into kubeedge/sedna

### Other information:
Recommended Skills:
- Fluent with Python
- Knowledge in MindSpore framework and understand one of the AI frameworks of sklearn, TensorFlow and Keras
- Have certain AI development experience and framework development experience

Related Communities
- [MindSpore](https://www.mindspore.cn/)
- [TinyMS](https://tinyms.readthedocs.io/en/latest/)
- [Sedna guide](https://sedna.readthedocs.io/en/latest/index/guide.html)
- [How to contribute Sedna](https://sedna.readthedocs.io/en/latest/contributing/prepare-environment.html)

### PR DESCRIPTION
**Pull request by Lj1ang**
**What type of PR is this?**
/kind feature

**What this PR does**:
This PR is a proposal that analyses Python SDK architecture of sedna and discusses the way to implement Python SDK support for TinyMS and demos of MindSpore and TinyMS

Fixes #307

### CODE REVIEW

File: docs/proposals/tinyms-support.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
+#  Implement Sedna Python SDK support TinyMS
+# 1. Project Background
```
Comment (jaypume) [inline docs/proposals/tinyms-support.md:2]: The headline should follow the existing proposal template, such as:
- Motivation
- Goals
- ...

File: docs/proposals/tinyms-support.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+Sedna can simply enable edge-cloud synergy capabilities to existing training and inference scripts, bringing the benefits of reducing costs, improving model performance, and protecting data privacy.
+
+## 1.2 Project Pain Points
+
+Currently, Sedna only supports sklearn's xgboost, Tensorflow, Keras and MindSpore, but doesn’t support MindSpore’s high-level API tool TinyMS.
+
+This project aims to implement Senda Python SDK ‘s support for TinyMS based on  Sedna's lib development interface and implements an Incremental Learning demo.
+
+## 1.3 Project Goals
```
Comment (jaypume) [inline docs/proposals/tinyms-support.md:15]: The benifit of TinyMS integration should be included in this section, not the target of OSPP.

File: docs/proposals/tinyms-support.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+- Implement Sedna Python SDK support of TinyMS
+- Design Incremental Learning demo
+- Write related documentation: usage documentation, use case reports
+- Submit the pull request to the repo
```
Comment (jaypume) [inline docs/proposals/tinyms-support.md:20]: This is the step of OSPP, it should be removed.

File: docs/proposals/tinyms-support.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+- Successfully merge to KubeEdge/Sedna
```
Comment (jaypume) [inline docs/proposals/tinyms-support.md:21]: This is the normal step of github coperation, it should be removed from this goal.

**Review by jaypume — CHANGES_REQUESTED** [review]

File: docs/proposals/tinyms-support.md (hunk 4 above)
Comment (Lj1ang) [reply docs/proposals/tinyms-support.md]: I have modified this proposal with the suggestions above

File: docs/proposals/tinyms-support.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
+- [Implement Sedna Python SDK support TinyMS](#mplement Sedna Python SDK support TinyMS)
```
Comment (jaypume) [inline docs/proposals/tinyms-support.md:1]: It shows error link when preview. Please fix it to make it preview correctly.
image

Comment (Lj1ang) [reply docs/proposals/tinyms-support.md]: It seems like that IDE is a little different from github in markdown. 
I have fixed this error.

Comment (MooreZheng) [reply docs/proposals/tinyms-support.md]: The index looks fine to me now

**Comment by MooreZheng** [comment]
The goals should highlight the benefit of mindspore, e.g., TinyMS, for the KubeEdge Community of SIG AI.

**Comment by Lj1ang** [comment]
> The goals should highlight the benefit of mindspore, e.g., TinyMS, for the KubeEdge Community of SIG AI.

Added

**Comment by jaypume** [comment]
> > The goals should highlight the benefit of mindspore, e.g., TinyMS, for the KubeEdge Community of SIG AI.
> 
> Added

I think more concret benifit can be put here, and you can refer to this page:
https://tinyms.readthedocs.io/en/latest/quickstart/overview.html#tinyms-project

**Comment by Lj1ang** [comment]
> > > The goals should highlight the benefit of mindspore, e.g., TinyMS, for the KubeEdge Community of SIG AI.
> > 
> > 
> > Added
> 
> I think more concret benifit can be put here, and you can refer to this page: https://tinyms.readthedocs.io/en/latest/quickstart/overview.html#tinyms-project

Added more concrete benefits.

**Review by jaypume — COMMENTED** [review]
It looks fine to me now.

**Pull request merged**